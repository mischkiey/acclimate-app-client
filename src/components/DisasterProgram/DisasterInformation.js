import React from 'react';

function DisasterInformation() {
    return (
        <>
        </>
    );
};

export default DisasterInformation;