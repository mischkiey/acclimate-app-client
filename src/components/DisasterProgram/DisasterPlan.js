import React from 'react';

function DisasterPlan() {
    return (
        <>
        </>
    );
};

export default DisasterPlan;