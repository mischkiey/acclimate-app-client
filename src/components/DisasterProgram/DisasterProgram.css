.disaster-program-group {
    border: 1px dashed #37474f;
    border-radius: 10px;
    display: flex;
    flex-flow: column;
    padding: 3px;
    margin: 3px;
    /* padding: 10px; */
    /* margin-bottom: 10px; */
}

/* button.item {
    align-self: flex-end;
} */

@media all and (min-width: 800px) {
    .disaster-program-group {
        border: 1px dashed #37474f;
        display: flex;
        flex-flow: column;
        padding: 5px;
        margin: 5px;
        /* padding: 10px; */
        /* margin-bottom: 10px; */
    }
}