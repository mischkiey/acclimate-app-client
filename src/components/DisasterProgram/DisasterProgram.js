import React, { Component } from 'react';
import './DisasterProgram.css';

class DisasterProgram extends Component {
    state = {
        expand: false,
    }

    renderPlanStepsByStage = (stage) => {
        return this.props.disaster_plan_steps.filter(step => step.disaster_plan_step_stage.toLowerCase() === stage).map(step => <li key={step.disaster_plan_step_id}>{step.disaster_plan_step}</li>);
    };

    handleToggleDisasterProgram = () => {
        this.setState({expand: !this.state.expand});
    };

    expand = () => {
        return (
            <div className='disaster-program-group'>
                <button className='item r-btn' onClick={() => this.handleToggleDisasterProgram()}><i className="material-icons">unfold_less</i></button>
                <article>
                    <h2 className='center'>Disaster Program</h2>
                    <p>{this.props.disaster_program_information}: Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.</p>
                </article>
                
                <article>
                    <h3>Readiness Stage</h3>
                    <ul>
                        {this.renderPlanStepsByStage('readiness')}
                    </ul>

                    <h3>Response Stage</h3>
                    <ul>
                        {this.renderPlanStepsByStage('response')}
                    </ul>

                    <h3>Recovery Stage</h3>
                    <ul>
                        {this.renderPlanStepsByStage('recovery')}
                    </ul>
                </article>
                <button className='item y-btn' onClick={() => this.props.handlePopulateLists(this.props.disaster_program_id)}><i className="material-icons">add_task</i></button>
                <button className='item r-btn' onClick={() => this.props.handleDeleteDisasterProgram(this.props.disaster_program_id)}><i className="material-icons">delete</i></button>
            </div>
        );
    };

    collapse = () => {
        return (
            <div className='disaster-program-group'>
                <button className='item y-btn' onClick={() => this.handleToggleDisasterProgram()}><i className="material-icons">unfold_more</i></button>
                <article>
                    <h2 className='center'>Disaster Program</h2>
                    <p>{this.props.disaster_program_information}: Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.</p>
                </article>
                <button className='item r-btn' onClick={() => this.props.handleDeleteDisasterProgram(this.props.disaster_program_id)}><i className="material-icons">delete</i></button>
            </div>
        );
    }

    render () {
        return (
            <>
                {
                    (this.state.expand)
                        ? this.expand()
                        : this.collapse()
                }
            </>
        );
    };
};

export default DisasterProgram;