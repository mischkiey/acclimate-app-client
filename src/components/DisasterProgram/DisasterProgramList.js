import React, {Component} from 'react';
import APIService from '../../services/api-services';
import TokenService from '../../services/token-services';
import { ExperimentalContext } from '../../contexts/ExperimentalContext';

import DisasterProgram from '../../components/DisasterProgram/DisasterProgram';

class DisasterProgramList extends Component {
    static contextType = ExperimentalContext;

    state = {
        programs: [],
        error: null,
    };

    handleAddToTaskList = (disaster_plan_step_itemable_shorthand) => {
        this.setState({error: null})

        const newUserTask = {
            user_task: disaster_plan_step_itemable_shorthand,
        };
        const token = TokenService.getAuthToken();

        return APIService.post('/disaster/user/task', newUserTask, token)
            .then(response => {
                this.context.tasks.push(response);
                const newTasks = this.context.tasks;
                this.context.setTasks(newTasks);
            })
            .catch(error => {
                this.setState({...error});
            });
    };

    handleAddToShoppingList = (disaster_plan_step_itemable_shorthand) => {
        this.setState({error: null})

        const newUserShoppingItem = {
            user_shopping_item: disaster_plan_step_itemable_shorthand,
        };
        const token = TokenService.getAuthToken();

        return APIService.post('/disaster/user/shopping', newUserShoppingItem, token)
            .then(response => {
                this.context.shoppingItems.push(response);
                const newShoppingItems = this.context.shoppingItems;
                this.context.setShoppingItems(newShoppingItems);
            })
            .catch(error => {
                this.setState({...error});
            });
    };

    handleDeleteDisasterProgram = (disaster_program_id) => {
        this.setState({error: null})

        const token = TokenService.getAuthToken();

        return APIService.del(`/disaster/user/program/${disaster_program_id}`, token)
            .then(() => {
                return APIService.get('/disaster/user/program', token)
                    .then(programs => {
                        this.setState({programs})
                    })
                    .catch(error => {
                        this.setState({error})
                    })
            })
            .catch(error => {
                this.setState({...error})
            });
    };

    componentDidMount() {
        this.setState({error: null})

        const token = TokenService.getAuthToken();

        return APIService.get('/disaster/user/program', token)
            .then(programs => {
                this.setState({programs})
            })
            .catch(({error}) =>{
                this.setState({error})
            });
    };

    render () {
        const programs = this.state.programs.map(program =>
            <DisasterProgram
                key={program.disaster_program_id}
                handleAddToTaskList={this.handleAddToTaskList}
                handleAddToShoppingList={this.handleAddToShoppingList}
                handleDeleteDisasterProgram={this.handleDeleteDisasterProgram}
                {...program}
            />
        )

        return (
            <section className='dashboard-component-group item'>
                {programs}
            </section>
        );
    };
};

export default DisasterProgramList;