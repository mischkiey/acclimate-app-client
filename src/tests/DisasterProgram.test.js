import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter, Router } from 'react-router-dom';
import DisasterProgram from '../components/DisasterProgram/DisasterProgram';

describe(`DisasterProgram Component`, () => {
    const disasterProgram = {
        disaster_program_information: 'About the blizzard disaster program',
        disaster_plan_steps: [
            {
                disaster_plan_step_id: 1, 
                disaster_plan_step: 'Blizzard readiness step', 
                disaster_plan_step_stage: 'Readiness',
            }
        ],
        disaster_id: 1,
    };

    describe(`Smoke test`, () => {
        it(`Renders without crashing`, () => {
            const div = document.createElement('div');
            ReactDOM.render(
                <BrowserRouter>    
                    <DisasterProgram {...disasterProgram} />
                </BrowserRouter>
                , div);
            ReactDOM.unmountComponentAtNode(div);
        });
    });
    
    describe(`Snapshot test`, () => {
        it(`Renders the UI as expected`, () => {
            const tree = renderer
                .create(
                    <BrowserRouter>
                        <DisasterProgram {...disasterProgram}/>
                    </BrowserRouter>
                )
                .toJSON()
            expect(tree).toMatchSnapshot()
        });
    });
})