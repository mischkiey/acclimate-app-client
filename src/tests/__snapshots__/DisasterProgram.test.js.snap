// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisasterProgram Component Snapshot test Renders the UI as expected 1`] = `
<div
  className="disaster-program-group"
>
  <button
    className="item y-btn"
    onClick={[Function]}
  >
    <i
      className="material-icons"
    >
      unfold_more
    </i>
  </button>
  <article>
    <h2
      className="center"
    >
      Disaster Program
    </h2>
    <p>
      About the blizzard disaster program
      : Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
    </p>
  </article>
  <button
    className="item r-btn"
    onClick={[Function]}
  >
    <i
      className="material-icons"
    >
      delete
    </i>
  </button>
</div>
`;
